sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env python3
+
+"""
+    stormond
+    Storage device Monitoring daemon for SONiC
+"""
+
+import os
+import signal
+import sys
+import threading
+import subprocess
+import shutil
+import json
+import time
+
+from sonic_py_common import daemon_base, device_info, logger
+from swsscommon import swsscommon
+from sonic_platform_base.sonic_storage.storage_devices import StorageDevices, BLKDEV_BASE_PATH
+
+#
+# Constants ====================================================================
+#
+
+# TODO: Once we no longer support Python 2, we can eliminate this and get the
+# name using the 'name' field (e.g., `signal.SIGINT.name`) starting with Python 3.5
+SIGNALS_TO_NAMES_DICT = dict((getattr(signal, n), n)
+                             for n in dir(signal) if n.startswith('SIG') and '_' not in n)
+
+SYSLOG_IDENTIFIER = "stormond"
+
+STORAGE_DEVICE_TABLE = "STORAGE_INFO"
+FSSTATS_SYNC_TIME_KEY = "FSSTATS_SYNC"
+
+FSIO_RW_JSON_FILE = "/usr/share/stormond/fsio-rw-stats.json"
```

Comment:
@assrinivasan don't we want this file to persist across sonic image upgrade too?

Reply:
Yes Prince -- that is why /host was chosen for the bindmount:

```
admin@str-s6100-acs-1:~$ ls -il /host/pmon/stormond/fsio-rw-stats.json
527881 -rw-r--r-- 1 root root 174 May 28 19:17 /host/pmon/stormond/fsio-rw-stats.json

admin@str-s6100-acs-1:~$ cat /host/pmon/stormond/fsio-rw-stats.json
{"sda": {"latest_fsio_reads": "20099", "latest_fsio_writes": "3559", "total_fsio_reads": "97865", "total_fsio_writes": "93334"}, "successful_sync_time": "1716923849.2595313"}admin@str-s6100-acs-1:~$

admin@str-s6100-acs-1:~$ sudo sonic-installer install -y http://10.201.148.43/pipelines/Networking-acs-buildimage-Official/broadcom/internal-202305/tagged/sonic-broadcom.bin
Downloading image...
...99%, 845 MB, 8574 KB/s, 0 seconds left...
efi not supported - exiting without verification

Installing image SONiC-OS-20230531.27 and setting it as default...
Command: bash /tmp/sonic_image
Verifying image checksum ... OK.
Preparing image archive ... OK.
Installing SONiC in SONiC
ONIE Installer: platform: x86_64-broadcom-r0
onie_platform: x86_64-dell_s6100_c2538-r0
Installing SONiC to /host/image-20230531.27
.
.
.
Command: sync

Command: sync

Command: sleep 3

Done
admin@str-s6100-acs-1:~$ sudo reboot
requested COLD shutdown
/var/log: 0 B (0 bytes) trimmed on /dev/loop1
/host: 5 GiB (5385273344 bytes) trimmed on /dev/sda4
Watchdog armed for 180 seconds
Wed May 29 02:29:10 AM UTC 2024 Issuing OS-level reboot ...
.
.
admin@str-s6100-acs-1:~$ ls -il /host/pmon/stormond/fsio-rw-stats.json
527881 -rw-r--r-- 1 root root 177 May 29 02:28 /host/pmon/stormond/fsio-rw-stats.json
admin@str-s6100-acs-1:~$
admin@str-s6100-acs-1:~$ cat /host/pmon/stormond/fsio-rw-stats.json
{"sda": {"latest_fsio_reads": "28627", "latest_fsio_writes": "36557", "total_fsio_reads": "126492", "total_fsio_writes": "129891"}, "successful_sync_time": "1716949729.9396956"}admin@str-s6100-acs-1:~$
admin@str-s6100-acs-1:~$
admin@str-s6100-acs-1:~$ sudo sonic-installer list
Current: SONiC-OS-20230531.27
Next: SONiC-OS-20230531.27
Available:
SONiC-OS-20230531.27
SONiC-OS-master-19070.557473-200645e4f
admin@str-s6100-acs-1:~$

```

Reply:
@assrinivasan add a comment ?
```
# Binds to /host/pmon/stormond/fsio-rw-stats.json
```

Reply:
Done in latest